server/api/min_resolved_ts.go

```diff
@@ -67,19 +70,37 @@ func (h *minResolvedTSHandler) GetStoreMinResolvedTS(w http.ResponseWriter, r *h
 	})
 }
 
-// @Tags     min_resolved_ts
-// @Summary  Get cluster-level min resolved ts.
+// @Tags         min_resolved_ts
+// @Summary      Get cluster-level min resolved ts.
+// @Description  Optionally, if a list of store IDs is provided in the request body,
+// it also returns the min_resolved_ts for the specified stores in a separate map.
 // @Produce  json
 // @Success  200  {array}   minResolvedTS
 // @Failure  500  {string}  string  "PD server failed to proceed the request."
-// @Router   /min-resolved-ts [get]
+// @Router       /min-resolved-ts [get]
 func (h *minResolvedTSHandler) GetMinResolvedTS(w http.ResponseWriter, r *http.Request) {
 	c := h.svr.GetRaftCluster()
 	value := c.GetMinResolvedTS()
 	persistInterval := c.GetPDServerConfig().MinResolvedTSPersistenceInterval
+
+	var storeMinResolvedTS map[uint64]uint64
+	if b, err := io.ReadAll(r.Body); err == nil && len(b) != 0 {
+		// stores ids is an optional parameter.
+		// if it is not empty, return the min resolved ts of the specified stores into map.
+		var ids []string
+		err = json.Unmarshal(b, &ids)
+		if err != nil {
+			h.rd.JSON(w, http.StatusBadRequest, err.Error())
+			return
+		}
+		c := h.svr.GetRaftCluster()
+		storeMinResolvedTS = c.GetMinResolvedTSByStoreIDs(ids)
+	}
+
 	h.rd.JSON(w, http.StatusOK, minResolvedTS{
-		MinResolvedTS:   value,
-		PersistInterval: persistInterval,
-		IsRealTime:      persistInterval.Duration != 0,
+		MinResolvedTS:      value,
+		PersistInterval:    persistInterval,
+		IsRealTime:         persistInterval.Duration != 0,
+		StoreMinResolvedTS: storeMinResolvedTS,
```

Comment:
It's strange that we have so many `min-resolved-ts` results returned in the same struct. Why not just reuse the `MinResolvedTS` field?

- If the store IDs are provided, `MinResolvedTS` field is the minimal resolved TS among them.
- If no store ID is provided, `MinResolvedTS` field is the minimal resolved TS among the whole cluster by default.

Reply:
> It's strange that we have so many min-resolved-ts results returned in the same struct. 

I am not getting this point, Do you mean `storeMinResolvedTS` is a redundant field? We need each store's ts to update client-go's `safeTS`.

> If the store IDs are provided, MinResolvedTS field is the minimal resolved TS among them.

u r right! this is more make sense.

Reply:
OK, I missed the point that we need the resolved TS for each store to update the client-go `safeTS`. It's fair to keep it, though the `MinResolvedTS` better keeps consistent with the given store IDs here.